### PR TITLE
Add the option to hide the generated datetime from the generated files

### PR DESCRIPTION
**Pull request by tvbeek**
On some projects I want to be sure that we didn't miss anything in the generated files. That is why I prefer to run the generate commands in the pipeline.

Because of the timestamps the files are always changed, with this PR it is possible to hide the datetime from the generated files so it is possible to see automated if there is any change found and a new file need to be committed.

### CODE REVIEW

File: config/ide-helper.php [hunk 1]
```diff
@@ -242,4 +242,8 @@
     */
     'include_class_docblocks' => false,
 
+    /**
+     * Hide the datetime from the generated files
+     */
```
Comment (mfn) [inline config/ide-helper.php:247]: the format of the comment looks nothing like the others in this file

File: resources/views/helper.php [hunk 2]
```diff
@@ -12,7 +12,7 @@
 
 /**
  * A helper file for Laravel, to provide autocomplete information to your IDE
- * Generated for Laravel <?= $version ?> on <?= date("Y-m-d H:i:s") ?>.
+ * Generated for Laravel <?= $version ?><?php if(!(isset($hide_datetime) && $hide_datetime)): ?> on <?= date("Y-m-d H:i:s") ?><?php endif;?>.
```
Comment (mfn) [inline resources/views/helper.php:15]: The PR always set the variable for the templates, so it seems to me `isset` isn't necessary?

Comment (mfn) [reply resources/views/helper.php:15]: Might I suggest a different approach?

In stead of "if not hide: echo $date; endif" how about the inverse / positive check:
- call the var `show_datetime`
- default to `true`
- write a "positive" check

```suggestion
 * Generated for Laravel <?= $version ?><?php if ($show_datetime) ?> on <?= date("Y-m-d H:i:s") ?><?php endif;?>.
```

Seems less cluttered and also more straight to the point.

(applies also for the check in the other file)

**Comment by tvbeek** [comment]
@mfn thanks for the feedback, I will wait on the answer from @barryvdh on your question about removing instead of adding another config. If that is clear I will update this PR.

**Comment by mfn** [comment]
Makes sense!

----
FTR, here are the reasons we've it in the first place:
- Basically since the beginning the date was added without man fanfare
  => https://github.com/barryvdh/laravel-ide-helper/commit/91033293b9fd879d1951838f913f0a6a3fbd852c
- Later the time was added:
  > Show time in generated files / I think this is useful for debugging.
  
  => https://github.com/barryvdh/laravel-ide-helper/pull/574

The "think … is …useful" sounds a bit fishy to me 😄 I think debugging should happen on different levels and favouring CI workflows in 2020 is more appropriate (aka just removing it).

I think this whole thing may also comes from a time when those files were "shared" online via gists for specific Laravel version, a practice I'd say shouldn't be endorsed at all anymore.

**Comment by mfn** [comment]
_Fun_ fact: I already mentioned that the `_ide_helper_models.php` does not include them.

In fact the test suite was recently refactored to use snapshots for the rewritten/generated PHP files and this wouldn't be possible if a date was written in them…

** :+1: ** for just removing them from my side

**Comment by barryvdh** [comment]
I think I added it because in the earliest versions, the actual gist was used more often. That's why the version and time were included. I don't see a benefit of including it now. Might as well remove it altogether.

**Comment by mfn** [comment]
Sent in https://github.com/barryvdh/laravel-ide-helper/pull/1000

**Comment by tvbeek** [comment]
> 
> 
> Sent in #1000

You are fast! 👍

**Comment by barryvdh** [comment]
Thanks

**Pull request closed without merging**